marshal/consumer.go

```diff
@@ -746,6 +745,51 @@ func (c *Consumer) Commit(msg *Message) error {
 	return cl.Commit(msg.Offset)
 }
 
+// Flush will cause us to upate all of the committed offsets. This operation can be
+// performed to periodically sync offsets without waiting on the internal flushing mechanism.
+func (c *Consumer) Flush() error {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	claims := make([]*claim, 0)
+	for topic, _ := range c.claims {
+		for partID, _ := range c.claims[topic] {
+			claims = append(claims, c.claims[topic][partID])
+		}
+	}
+
+	// Do flushing concurrently because they involve sending messages to Kafka
+	// which can be slow if done serially
+	waiter := &sync.WaitGroup{}
+	waiter.Add(len(claims))
+	errChan := make(chan error, len(claims))
+
+	for _, cl := range claims {
+		cl := cl
+		go func() {
+			defer waiter.Done()
+			if err := cl.Flush(); err != nil {
+				errChan <- err
+			}
+		}()
+	}
+
+	// Wait for all flushes to finish
+	waiter.Wait()
```

Comment:
As discussed offline: Agreed that this has the potential for never returning but I think that is technically valid for a flush/fsync operation. If the caller cares about timeouts they can add one on their call.